cameo/flux_analysis/analysis.py

```diff
@@ -674,24 +705,15 @@ def mass_yield(self):
         float
             gram product per 1 g of feeding source or nan if more than one product or feeding source
         """
-        not_unique = (len(self.source.metabolites) != 1,
-                      len(self.product_reaction.metabolites) != 1)
-        if any(not_unique):
+        if self.source is None:
             return numpy.nan
         try:
-            source_flux = self.source.flux
-            product_flux = self.product_reaction.flux
-        except AssertionError:
+            source, source_flux = single_flux(self.source, consumption=True)
+            product, product_flux = single_flux(self.product_reaction, consumption=False)
+        except ValueError:
             return numpy.nan
-        source_metabolite = list(self.source.metabolites)[0]
-        product_metabolite = list(self.product_reaction.metabolites)[0]
-        product_mass = product_metabolite.formula_weight
-        source_mass = source_metabolite.formula_weight
-        try:
-            mol_prod_mol_src = product_flux / (source_flux * -1)
-            return (mol_prod_mol_src * product_mass) / source_mass
-        except ZeroDivisionError:
-            return 0
+        mol_prod_mol_src = product_flux / source_flux
```

Comment:
Can't remember if that's default in Python 2.7 now but to be safe you might want to add `from __future__ import division` to the top otherwise Python 2 will do integer division.